sycl/plugins/cuda/CMakeLists.txt

```diff
@@ -24,51 +23,19 @@ else()
   )
 endif()
 
-add_library(pi_cuda SHARED
-  "${sycl_inc_dir}/CL/sycl/detail/pi.h"
-  "${sycl_inc_dir}/CL/sycl/detail/pi.hpp"
-  "pi_cuda.hpp"
-  "pi_cuda.cpp"
-)
-
-add_dependencies(sycl-toolchain pi_cuda)
-
-set_target_properties(pi_cuda PROPERTIES LINKER_LANGUAGE CXX)
-
-target_include_directories(pi_cuda
-  PRIVATE
-    ${sycl_inc_dir}
-)
-
-target_link_libraries(pi_cuda
-  PRIVATE
+add_sycl_plugin(cuda
+  SOURCES
+    "${sycl_inc_dir}/CL/sycl/detail/pi.h"
+    "${sycl_inc_dir}/CL/sycl/detail/pi.hpp"
+    "pi_cuda.hpp"
```

Comment:
Why header files are listed as SOURCES?

Reply:
AFAIK, people do that to support Visual Studio projects generation. I just copied those as-is.